### PR TITLE
Move config back into amethyst_config

### PR DESCRIPTION
**Pull request by Aceeri**
Adds a bit more portability (can be used outside of amethyst) and no longer requires importing your own `serde`/`serde_derive`/`serde_yaml`/`toml` to use it.

### CODE REVIEW

File: src/config/Cargo.toml [hunk 1]
```diff
@@ -15,4 +15,6 @@ appveyor = { repository = "amethyst/amethyst" }
 travis-ci = { repository = "amethyst/amethyst" }
```
Comment (ebkalderon) [inline src/config/Cargo.toml]: Before we push out the new `amethyst_config` release, please replace this:
```
appveyor = { repository = "amethyst/amethyst" }
```
with this:
```
appveyor = { repository = "amethyst/amethyst", branch = "develop" }
```

**Comment by Aceeri** [comment]
Going to wait to merge this until we come to a conclusion on what we want the directory structure to be.

File: Cargo.toml [hunk 2]
```diff
@@ -25,6 +25,7 @@ profiler = ["thread_profiler/thread_profiler"]
 
 [dependencies]
 amethyst_renderer = { path = "src/renderer", version = "0.4.3" }
+amethyst_config = { path = "src/config", version = "0.3.0" }
```
Comment (ebkalderon) [inline Cargo.toml]: Move this up one line so crate dependencies remain in alphabetical order.

**Comment by ebkalderon** [comment]
Alright, I think we can merge this now.

**Pull request merged**